lib/Sema/CSSimplify.cpp

```diff
@@ -2070,6 +2070,14 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
       if (path[lastIdx - 1].is<LocatorPathElt::PatternMatch>() &&
           path[lastIdx].is<LocatorPathElt::FunctionArgument>())
         return true;
+      
+      if (path.size() > 5) {
```

Comment:
Why `5`? Should this logic be more dynamic e.g. find the index for `LocatorPathElt::PatternMatch` and then look through function arguments and tuple elements from that index forward instead of using specific indexes like `3` or `4`? To me even seems a more stable approach for `N` levels of nesting and we could even merge the logic above in there as well.
Note I'm not a reviewer so this is more an opinion that maybe is worth considering =]

Reply:
Thanks, @LucianoPAlmeida :) . I totally agree with you. I will try to fix it.

Reply:
I wonder if simply checking that there is a `PatternMatch` element is in the path is good enough here to determine that location belongs to a pattern matching context.

Reply:
@xedin 
Would it make more sense to define a static value like the current fix?
Or is it that the current fix looks weird?

I tried to change it to use dynamic variables a while ago, but the current modification seemed to be fine considering the case where the pattern matching element was not included.

Reply:
These constants are usually a sign that something is not right. I am wondering whether iterating over a path to find first `PatternMatch` element would actually be sufficient here instead of all the checks we currently have which seem all be too narrow.

Reply:
> I am wondering whether iterating over a path to find first `PatternMatch` element would actually be sufficient here instead of all the checks we currently have which seem all be too narrow.

That makes sense. I was also noting that we are dropping `TupleType` information from locator path but not `TupleElement`. Either way, if we only find first `PatternMatch` we can remove that logic entirely. Even if the narrowing down to `LocatorPathElt::FunctionArgument` is needed we can check the next index after finding the last `PatternMatch` position in the path which also makes this dropping not needed.

Reply:
Sounds like a plan!

Reply:
Sorry for the late reply.
Thank you everyone for checking and following me.
I tried to fix it, so please check it.

Reply:
Sorry for the late reply.
Thank you everyone for checking and following me.
I tried to fix it, so please check it.